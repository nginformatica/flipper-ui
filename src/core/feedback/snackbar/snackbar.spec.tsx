import React from 'react'
import { act } from 'react-dom/test-utils'
import { render, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import Snackbar from '@/test/mocks/snackbar-mock'

describe('Snackbar', () => {
    it('should render a Snackbar', async () => {
        render(<Snackbar snackProps={{ message: 'Snackbar Message' }} />)

        const btn = screen.getByTestId('snackbar-button')

        await act(async () => {
            await userEvent.click(btn)
        })

        const message = screen.getByText('Snackbar Message')

        expect(message).toBeDefined()
    })

    it('should render with action', async () => {
        const onClose = jest.fn()

        render(
            <Snackbar
                withAction
                snackProps={{ message: 'Snackbar Message', onClose }}
            />
        )

        const btn = screen.getByTestId('snackbar-button')

        await act(async () => {
            await userEvent.click(btn)
        })

        const closeBtn = screen.getByRole('close-icon-button')

        await act(async () => {
            await userEvent.click(closeBtn)
        })

        await waitFor(() => expect(onClose).toHaveBeenCalled())
    })

    it('should call onClick', async () => {
        const onClick = jest.fn()

        render(
            <Snackbar
                snackProps={{
                    message: 'Snackbar Message',
                    autoHide: 100,
                    onClick
                }}
            />
        )

        const btn = screen.getByTestId('snackbar-button')

        await act(async () => {
            await userEvent.click(btn)
        })

        const message = screen.getByText('Snackbar Message')

        await act(async () => {
            await userEvent.click(message)
        })

        await waitFor(() => expect(onClick).toHaveBeenCalled())
    })

    it('should call onClose', async () => {
        const onClose = jest.fn()

        render(
            <Snackbar
                snackProps={{
                    message: 'Snackbar Message',
                    autoHide: 100,
                    onClose
                }}
            />
        )

        const btn = screen.getByTestId('snackbar-button')

        await act(async () => {
            await userEvent.click(btn)
        })

        const closeBtn = screen.getByRole('close-icon-button')

        await act(async () => {
            await userEvent.click(closeBtn)
        })

        await waitFor(() => expect(onClose).toHaveBeenCalled())
    })

    it('should match snapshot', () => {
        const onClick = jest.fn()
        const onClose = jest.fn()

        const { container } = render(
            <Snackbar
                withAction
                snackProps={{
                    message: 'Snackbar Message',
                    onClick,
                    onClose
                }}
            />
        )

        expect(container).toMatchSnapshot()
    })
})
