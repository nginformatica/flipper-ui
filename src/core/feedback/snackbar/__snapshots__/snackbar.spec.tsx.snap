// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="snackbar-button"
    href=""
    style="opacity: 1;"
    tabindex="0"
    type="button"
  >
    Open
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
